crates/sui-adapter/src/adapter.rs

```diff
@@ -180,10 +188,14 @@ fn execute_internal<
             args,
             gas_status.get_move_gas_status(),
         )
-        .and_then(|ret| Ok((ret, session.finish()?)))?;
+        .and_then(|ret| Ok((ret, session.finish_with_extensions()?)))?;
+    debug_assert!(return_values.is_empty());
```

Comment:
> Should this be a non-debug assert!? 

We typically do not use those ever in Move, though I have been unsure in Sui. If the check is critical, it needs to be an error, because the assert will cause a panic and take down the entire Rust instance the VM is embedded in. 

But 
> Because if not, it could offer a way to effectively drop a non-drop value, by returning it out of an entry fun, right?

For this reason, it really should be an invariant violation with a full error as I described above.

Reply:
> Do we have something in place to prevent the creation of an entry function with a return value?

The entry point verifier for Sui

Reply:
> This seems like a useful check, but it's unrelated to the change at hand, right? (Just checking my own understanding.) Seeing as it's here though, might as well review it:

Yeah I just through it in here but it probably shouldnt have been added with this PR if I was trying to keep things minimal